File: python/loccount/interactors.py
from commandline import get_path
from filenameprovider import find_source_filenames
from fileprovider import read_file
from linesofcode import create_locstat


def start(args, on_locstat, on_finished):
    path = get_path(args)
    find_source_filenames(path, lambda filename: on_file(filename, on_locstat), on_finished)


def on_file(filename, on_locstat):
    lines = read_file(filename)
    locstat = create_locstat(filename, lines)
    on_locstat(locstat)

File: java/loccount/src/de/lieser_online/Interactors.java
package de.lieser_online;

import java.util.List;
import java.util.function.BiConsumer;
import java.util.function.Consumer;

public class Interactors {
    public static void start(String[] args, Consumer<LocStat> onLocStat, Runnable onFinished) {
        String path = CommandLine.getDirectory(args);
        FilesystemProvider.findSourceFilenames(path,
            filename -> {
                List<String> lines = CodefileProvider.readFile(filename);
                LocStat locstat = Loc.countLines(filename, lines);
                onLocStat.accept(locstat);
            },
            () -> {
                onFinished.run();
            });
    }
}
